README.md

```diff
@@ -190,11 +190,488 @@ autotools-snap --files ./components/**/*.meta.ts
 
+#### Snap Configuration
+
+Configuration is as simple as adding a `autotools.config.js` file in your `.autotools` folder.
+If you wish to use a configuration file named something different, or located somewhere else, you can specify a specific file using the `--config` flag, like so: 
+
+`yarn snap --config ./config/myConfig.js`
+
+The format of the `autotools.config.js` file is as follows:
+
+``` js
+module.exports = {
+    applitoolsApiKeyName: ‘WHATEVER_YOU_NAMED_YOUR_ENV_VARIABLE’,
+    projectHooks: [],
+    componentHooks: [],
+    simHooks: [],
+    afterHooks: []
+}
+```
+All fields are optional. However, without hooks, Snap won't do anything.
+
+#### Building
+
+Each project may have a different or unique build step. Components may be Stylable-based, or use Sass, or css-modules, or have special imports, or be written in Typescript... the list goes on. In order to keep Snap generic, and usable in many different situations, we've implemented what we're calling *hooks*.
+
+Hooks allow a project to build itself, and then give Snap the information it needs in order to provide screenshot regression tests.
+
+This is our contract, in its simplest terms: Snap will call all of the hooks in the config in a specific order, passing information about the project, and after all the hooks have been called, Snap expects to receive a list containing the locations of HTML files which need to be sent to Applitools. Luckily, we already provide some hooks that cover common-use cases, so as a project, you may not need to write your own.
+
+##### The Hooks
+
+Snap exposes four hooks (similar in idea to webpack plugins), which allow behaviour to be implemented during different stages of Snap's execution:
+
+| Hook       | Provided by Snap                                                | Returned by Hook            |
+|------------|-----------------------------------------------------------------|-----------------------------|
+| Project    | `snapInfo: ISnapInfo`                                           | `Promise<any>`              |
+| Component  | `compSnapInfo: ICompSnapInfo`                                   | `Promise<IFileInfo \| void>` |
+| Simulation | `simSnapInfo: ISimSnapInfo`                                     | `Promise<IFileInfo \| void>` |
+| After      | `snapInfo: ISnapInfo, projectHookData: any, files: IFileInfo[]` | `Promise<void>`             |
+
+- **project**: called once, at the beginning. This is for setup (creating temporary directories, building the project if it only needs to be built once, etc.) 
+- **component**: called once for every component in the project which has a metadata file.
+- **sim**: called once for each simulation of every component.
+- **after**: called once after all other hooks have been called. This is where you'd do any necessary cleanup (deleting temporary directories, files, etc.)
+
+The interface for `IFileInfo` is as follows:
+
+```ts
+interface IFileInfo {
```

Comment:
maybe add inline comments describing each attribute? the example isn't enough i feel